src/asImageWidthSrcSet.ts

```diff
@@ -0,0 +1,102 @@
+import { ImageFieldImage } from "@prismicio/types";
+import {
+	buildURL,
+	buildWidthSrcSet,
+	BuildWidthSrcSetParams,
+} from "imgix-url-builder";
+
+import { imageThumbnail as isImageThumbnailFilled } from "./isFilled";
+
+/**
+ * The return type of `asImageWidthSrcSet()`.
+ */
+type AsImageWidthSrcSetReturnType<Field extends ImageFieldImage> =
+	Field extends ImageFieldImage<"empty">
+		? null
+		: {
+				/**
+				 * The Image field's image URL with Imgix URL parameters (if given).
+				 */
+				src: string;
+
+				/**
+				 * A width-based `srcset` attribute value for the Image field's image
+				 * with Imgix URL parameters (if given).
+				 */
+				srcset: string;
+		  };
+
+/**
+ * Creates a width-based `srcset` from an Image field with optional image
+ * transformations (via Imgix URL parameters).
+ *
+ * If the Image field contains responsive views, each responsive view is used as
+ * a width in the resulting `srcset`.
+ *
+ * If a `widths` parameter is not given, `[400, 800, 1600]` will be used by default.
+ *
+ * @example
+ *
+ * ```ts
+ * const srcset = asImageWidthSrcSet(document.data.imageField, {
+ * 	widths: [400, 800, 1600],
+ * 	sat: -100,
+ * });
+ * // => {
+ * //   src:    'https://images.prismic.io/repo/image.png?sat=-100',
+ * //   srcset: 'https://images.prismic.io/repo/image.png?sat=-100&width=400 400w, ' +
+ * //           'https://images.prismic.io/repo/image.png?sat=-100&width=800 800w,' +
+ * //           'https://images.prismic.io/repo/image.png?sat=-100&width=1600 1600w'
+ * // }
+ * ```
+ *
+ * @param field - Image field (or one of its responsive views) from which to get
+ *   an image URL.
+ * @param params - An object of Imgix URL API parameters. The `widths` parameter
+ *   defines the resulting `srcset` widths.
+ *
+ * @returns A `srcset` attribute value for the Image field with Imgix URL
+ *   parameters (if given). If the Image field is empty, `null` is returned.
+ * @see Imgix URL parameters reference: https://docs.imgix.com/apis/rendering
+ */
+export const asImageWidthSrcSet = <Field extends ImageFieldImage>(
+	field: Field,
+	params: Omit<BuildWidthSrcSetParams, "widths"> &
+		Partial<Pick<BuildWidthSrcSetParams, "widths">> = {},
+): AsImageWidthSrcSetReturnType<Field> => {
+	if (isImageThumbnailFilled(field)) {
+		const { widths = [400, 800, 1600], ...urlParams } = params;
```

Comment:
I remember you mentioning this default, but is it backed by anything?

If not, out of the blue, I'd have a tendency to use Tailwind's breakpoint for example: https://tailwindcss.com/docs/responsive-design (which were based on common devices), so something like `[640, 768, 1024, 1280]`

Reply:
I thought I took those numbers from `gatsby-image`'s default breakpoints, but I don't see that in their docs. I might have just made those up in my head.

Tailwind's sizes seem quite small at the top end for images. Those breakpoints work well for responsive design, but don't take into account images that could span 100% viewport width. For reference, 1280px is half of what my screen uses (2560px, actually 5120px since it's rendered at 2x).

We could go with a simplified version of [Next.js's image breakpoints](https://nextjs.org/docs/api-reference/next/image#device-sizes): `[640, 750, 828, 1080, 1200, 1920, 2048, 3840]`

This is what every other value would look like: `[640, 828, 1200, 2048, 3840]`

Reply:
I implemented all of Next.js's default breakpoints. If this output seems too verbose to you, I can change it to the trimmed down version.

Reply:
Works for me :+1:

Reply:
Done ✅